django_select2/forms.py

```diff
@@ -245,14 +249,16 @@ def build_attrs(self, base_attrs, extra_attrs=None):
         }
 
         if self.dependent_fields:
-            default_attrs['data-select2-dependent-fields'] = " ".join(self.dependent_fields)
+            default_attrs['data-select2-dependent-fields'] = (
+                " ".join(self.dependent_fields)
+            )
 
         default_attrs.update(base_attrs)
 
         attrs = super().build_attrs(default_attrs, extra_attrs=extra_attrs)
 
-        # encrypt instance Id
-        self.widget_id = signing.dumps(id(self))
+        self.uuid = str(uuid.uuid4())
```

Comment:
This should probably set during `__init__` and not any other method, otherwise rendering the same form twice will have strange side effects.